backend/GymBeam/Controllers/ReservationController.cs

```diff
@@ -39,12 +43,58 @@ public async Task<ActionResult<IEnumerable<ReservationResponse>>> Get()
         }
 
         [HttpPost]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
 #if !DEBUG
         [Authorize(Roles = Roles.User)]
 #endif
-        public IActionResult Post([FromBody] ReservationRequest dto)
+        public async Task<ActionResult> Post([FromBody] ReservationRequest dto)
         {
-            return NoContent();
+            int userId;
+            var request = new CreateReservationCommand
+            {
+                ActivityId = dto.ActivityId,
+                StartTime = dto.StartTime
+            };
+            try
+            {
+                if (!Request.Cookies.TryGetValue(Cookies.UserId, out string cookiesUserId))
+                    throw new InvalidCookieException(Cookies.UserId);
+                if (!int.TryParse(cookiesUserId, out userId))
+                    throw new InvalidUserIdException();
+
+                request.UserId = userId;
```

Comment:
I'm not sure if assigning request.UserId this way is correct

Reply:
better to have is from request, maybe admin wants to reserve?

Reply:
we need to check if user that sends this post is user that is logged in or admin.
As a user only me or admin can change my reservations.
I would add UserId to reservation request and to reservation command
and add LoggedUserId to only reservation command
So Command will have both UserId(from request) and LoggedUserId(from cookies) but Request will have only UserId.
This way in Command handler you can check whether User is logged user or admin